db/migrate/20220304192308_create_rule_group_relationships.rb

```diff
@@ -0,0 +1,10 @@
+class CreateRuleGroupRelationships < ActiveRecord::Migration[7.0]
+  def change
+    create_table :rule_group_relationships, id: :uuid do |t|
+      t.references :left, type: :uuid, polymorphic: true
+      t.references :right, type: :uuid, polymorphic: true
+      t.string :relationship
+    end
+    add_index(:rule_group_relationships, %i[left_id right_id relationship], name: 'index_rule_group_relationships_on_polymorphic_columns', unique: true)
```

Comment:
Do we really need to set the name explicitly?

Reply:
Also move it as `t.index` inside the block.

Reply:
I was getting the error message saying that the index name was too long so I had to set it to something shorter